src/main/java/de/tum/in/www1/artemis/service/WebsocketMessagingService.java

```diff
@@ -40,9 +48,46 @@ public void broadcastNewResult(Participation participation, Result result) {
         var originalParticipation = result.getParticipation();
         result.setParticipation(originalParticipation.copyParticipationId());
 
-        messagingTemplate.convertAndSend("/topic/participation/" + participation.getId() + "/newResults", result);
+        // TODO: Are there other cases that must be handled here?
+        if (participation instanceof StudentParticipation) {
+            StudentParticipation studentParticipation = (StudentParticipation) participation;
+            studentParticipation.getStudents().forEach(user -> messagingTemplate.convertAndSendToUser(user.getLogin(), NEW_RESULT_TOPIC, result));
+        }
+
+        // Send to tutors, instructors and admins
+        messagingTemplate.convertAndSend(getResultDestination(participation.getExercise().getId()), result);
 
         // recover the participation because we might want to use it again after this method
         result.setParticipation(originalParticipation);
     }
+
+    /**
+     * Returns true if the given destination should be handled by this service
```

Comment:
Does not really explain what the method is actually checking.

Reply:
You are right, I've adjusted the documentation.

Reply:
@madwau 
> Left some code comments.

Thanks for the feedback, I addressed the comments.

Reply:
Great!